context.go

```diff
@@ -225,11 +260,29 @@ func (context *Context) Close() {
 
 // TotalRuntime returns the cumulated WAF runtime across various run calls within the same WAF context.
 // Returned time is in nanoseconds.
-func (context *Context) TotalRuntime() (overallRuntimeNs, internalRuntimeNs uint64) {
-	return context.totalOverallRuntimeNs.Load(), context.totalRuntimeNs.Load()
+// Deprecated: use Timings instead
+func (context *Context) TotalRuntime() (uint64, uint64) {
+	return uint64(context.timings.RunNs.Load() * int64(time.Nanosecond)), uint64(context.timings.RuntimeNs.Load())
 }
 
 // TotalTimeouts returns the cumulated amount of WAF timeouts across various run calls within the same WAF context.
+// Deprecated: use Timings instead
 func (context *Context) TotalTimeouts() uint64 {
-	return context.timeoutCount.Load()
+	return context.timeBudgetExpirationCount.Load()
+}
+
+// Timings returns the cumulative time spent in various parts of the WAF, all in nanoseconds
+// and the timeout value used
+func (context *Context) Timings() Timings {
+	return context.timings
+}
+
+// SetTimeBudget sets the maximum time the WAF is allowed to run for the given context cumulatively on all run calls.
+func (context *Context) SetTimeBudget(timeBudget time.Duration) {
+	context.timings.TimeBudget = timeBudget
+}
```

Comment:
What is the use-case for being able to change the `TimeBudget` after the context has been created? It seems to me this field is expected to be immutable?